sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -365,6 +368,16 @@ def translate_output_properties(res: 'Resource', output: Any) -> Any:
     return output
 
 
+def contains_unknowns(val: Any) -> bool:
+    if known_types.is_unknown(val):
+        return True
+    if isinstance(val, dict):
+        return any(map(contains_unknowns, val.values()))
+    if isinstance(val, list):
+        return any(map(contains_unknowns, val))
+    return False
```

Comment:
duplicated?

Reply:
Intentionally, and with a very very slightly different implementation (note the was we check for an instance of `Unknown`). Python strictly disallows import cycles, so we can't use the implementation in the public API.